### PR TITLE
Mesh 2045/improve permissions

### PR DESCRIPTION
**Pull request by HenriqueNogara**
This PR adds the `portfolio_kind` parameter to the  `issue` extrinsic and changes `issue` and `controller_transfer` not to require custody checks.

### modified API

- Adds a `portfolio_kind` parameter to the `issue` extrinsic;

### modified logic

-  `issue` and `controller_transfer` no longer require custody checks;

### CODE REVIEW

File: pallets/common/src/traits/asset.rs [hunk 1]
```diff
@@ -17,108 +17,21 @@ use frame_support::decl_event;
...
+    AssetIdentifier, Balance, Document, DocumentId, IdentityId, PortfolioId, PortfolioKind,
+    ScopeId, Ticker,
 };
-use sp_std::prelude::Vec;
 
 use crate::traits::nft::NFTTrait;
 use crate::traits::{checkpoint, compliance_manager, external_agents, portfolio, statistics};
 
-/// This trait is used by the `identity` pallet to interact with the `pallet-asset`.
-pub trait AssetSubTrait {
```
Comment (HenriqueNogara) [inline pallets/common/src/traits/asset.rs:36]: This was just moved to the end of the file.

File: pallets/asset/src/lib.rs [hunk 2]
```diff
@@ -932,143 +926,52 @@ decl_error! {
     }
 }
 
-impl<T: Config> AssetFnTrait<T::AccountId, T::RuntimeOrigin> for Module<T> {
```
Comment (HenriqueNogara) [inline pallets/asset/src/lib.rs:997]: Both trait impl were moved to the end of the file. No changes made

File: pallets/asset/src/benchmarking.rs [hunk 3]
```diff
@@ -353,7 +353,7 @@ benchmarks! {
 
     issue {
         let (owner, ticker) = owned_ticker::<T>();
-    }: _(owner.origin, ticker, (1_000_000 * POLY).into())
+    }: _(owner.origin, ticker, (1_000_000 * POLY).into(), PortfolioKind::Default)
```
Comment (Neopallium) [inline pallets/asset/src/benchmarking.rs]: Better to use a non-default portfolio in the benchmarks.

File: pallets/nft/src/lib.rs [hunk 4]
```diff
@@ -329,8 +330,12 @@ impl<T: Config> Module<T> {
             CollectionTicker::try_get(&ticker).map_err(|_| Error::<T>::CollectionNotFound)?;
 
         // Ensure origin is agent with custody and permissions for default portfolio.
```
Comment (Neopallium) [inline pallets/nft/src/lib.rs]:
```suggestion
        // Ensure origin is agent with custody and permissions for portfolio.
```

**Review by Neopallium — CHANGES_REQUESTED** [review]
Very minor changes, otherwise looks great.

**Review by Neopallium — APPROVED** [review]

**Review by adamdossa — APPROVED** [review]

**Pull request merged**